// @docker/docker-mui-theme/dist/themes.js
import { alpha, createTheme } from '@mui/material';
import createPalette from '@mui/material/styles/createPalette';
import createTypography from '@mui/material/styles/createTypography';
// see https://fontsource.org/docs/variable-fonts
import '@fontsource/open-sans/variable.css';
import { colors } from './colors';
const primaryColor = {
    light: '#6BC3F3',
    main: colors.marine50,
    dark: '#096BD4',
};
function createDDTheme(themeSpecificPalette) {
    const pal = createPalette({
        primary: {
            light: primaryColor.light,
            main: primaryColor.main,
            dark: primaryColor.dark,
        },
        secondary: {
            main: '#E12B0C',
            dark: '#871907',
        },
        action: {
            hover: alpha(primaryColor.light, 0.4),
            selected: alpha(primaryColor.light, 0.25),
        },
        ...themeSpecificPalette,
    });
    const typography = createTypography(pal, {
        fontFamily: '"Open SansVariable", "Helvetica Neue", sans-serif',
        fontSize: 12,
        subtitle1: {
            fontSize: 16,
        },
        body1: {
            fontSize: 14,
        },
        subtitle2: {
            fontSize: 22,
            fontWeight: 600,
        },
    });
    return createTheme({
        palette: pal,
        components: {
            MuiAppBar: {
                styleOverrides: {
                    colorPrimary: {
                        backgroundColor: '#27343B',
                    },
                },
            },
            MuiButton: {
                styleOverrides: {
                    root: {
                        textTransform: 'initial',
                        fontWeight: 600,
                    },
                    contained: {
                        boxShadow: 'none',
                        '&:hover, &:focus': {
                            boxShadow: 'none',
                        },
                    },
                    outlinedPrimary: {
                        color: pal.primary.main,
                        border: `1px solid ${pal.primary.main}`,
                    },
                    outlinedSecondary: {
                        color: pal.secondary.main,
                        border: `1px solid ${pal.secondary.main}`,
                    },
                },
            },
            MuiIconButton: {
                styleOverrides: {
                    root: {
                        transition: 'color 0.3s ease-out',
                        '&:hover': {
                            backgroundColor: 'none',
                        },
                    },
                },
            },
            MuiCheckbox: {
                defaultProps: {
                    disableRipple: true,
                },
                styleOverrides: {
                    root: {
                        paddingTop: 4,
                        paddingBottom: 4,
                    },
                    colorPrimary: {
                        '&:hover': {
                            backgroundColor: 'transparent',
                        },
                        '&$checked': {
                            '&:hover': {
                                backgroundColor: 'transparent',
                            },
                        },
                    },
                },
            },
            MuiSwitch: {
                styleOverrides: {
                    colorPrimary: {
                        '&:hover': {
                            backgroundColor: 'transparent',
                        },
                        '&$checked': {
                            '&:hover': {
                                backgroundColor: 'transparent',
                            },
                        },
                    },
                },
            },
            MuiChip: {
                styleOverrides: {
                    labelSmall: {
                        paddingLeft: 4,
                        paddingRight: 4,
                    },
                    sizeSmall: {
                        height: 14,
                        fontSize: 8,
                        borderRadius: 4,
                        textTransform: 'uppercase',
                    },
                },
            },
            MuiFormLabel: {
                styleOverrides: pal.mode === 'dark'
                    ? {
                        root: {
                            color: '#B7BBBE',
                            '&$disabled': {
                                color: '#70787D',
                            },
                            '&$focused': {
                                color: pal.primary.main,
                            },
                        },
                    }
                    : {},
            },
            MuiInput: {
                styleOverrides: {
                    input: {
                        '&:focus': {
                            backgroundColor: pal.mode === 'dark'
                                ? alpha(colors.white0, 0.1)
                                : alpha('#000000', 0.03),
                        },
                    },
                    underline: {
                        '&:after': {
                            borderBottomColor: pal.primary.main,
                        },
                    },
                },
            },
            MuiInputLabel: {
                styleOverrides: {
                    shrink: {
                        transform: 'translate(0, 1.6px) scale(0.86)',
                    },
                },
            },
            MuiListItem: {
                styleOverrides: {
                    root: {
                        paddingTop: 6,
                        paddingBottom: 6,
                        '&$selected': {
                            backgroundColor: alpha(primaryColor.light, 0.2),
                        },
                    },
                    button: {
                        '&:hover': {
                            backgroundColor: alpha(primaryColor.light, 0.1),
                        },
                    },
                },
            },
            MuiTooltip: {
                styleOverrides: {
                    tooltip: {
                        fontSize: '10px',
                        lineHeight: '13px',
                        letterSpacing: '1px',
                        textTransform: 'uppercase',
                        color: 'white',
                        backgroundColor: pal.mode === 'dark' ? '#5C6972' : '#2F3B43',
                    },
                },
            },
            MuiListItemIcon: {
                styleOverrides: {
                    root: {
                        minWidth: 36,
                    },
                },
            },
            MuiMenu: {
                styleOverrides: {
                    paper: {
                        background: pal.mode === 'dark' ? '#303E46' : colors.white5,
                        boxShadow: 'none',
                    },
                },
            },
            MuiMenuItem: {
                styleOverrides: {
                    root: {
                        minHeight: 'auto',
                        '&:focus': {
                            '& .MuiListItemIcon-root, & .MuiListItemText-primary': {
                                color: 'white',
                            },
                        },
                    },
                    gutters: {
                        fontSize: typography.body1.fontSize,
                    },
                },
            },
            MuiRadio: {
                styleOverrides: {
                    root: {
                        paddingTop: 4,
                        paddingBottom: 4,
                    },
                },
            },
            MuiFilledInput: {
                styleOverrides: {
                    root: {
                        borderRadius: '0 !important',
                        backgroundColor: pal.mode === 'dark' ? alpha(colors.white0, 0.09) : '#ECECEC',
                        '&:focus, &:hover': {
                            backgroundColor: pal.mode === 'dark' ? alpha(colors.white0, 0.13) : '#E4E4E4',
                        },
                    },
                    underline: {
                        '&:after': {
                            borderBottomColor: pal.primary.main,
                        },
                    },
                },
            },
            MuiOutlinedInput: {
                styleOverrides: {
                    notchedOutline: {
                        borderColor: pal.mode === 'dark'
                            ? '#555D61 !important'
                            : `${alpha('#000000', 0.23)} !important`,
                        borderWidth: '1px !important',
                    },
                    root: {
                        '&$disabled': {
                            backgroundColor: pal.mode === 'dark' ? '#2C373E' : '#EFEFEF',
                            borderRadius: '4px',
                            color: alpha('#70787D', 0.5),
                        },
                    },
                },
            },
            MuiPaper: {
                styleOverrides: {
                    root: {
                        borderRadius: '0 !important',
                    },
                },
            },
            MuiList: {
                styleOverrides: {
                    root: {
                        paddingTop: '0 !important',
                        paddingBottom: '0 !important',
                    },
                },
            },
            MuiTableRow: {
                styleOverrides: {
                    hover: {
                        backgroundColor: 'transparent !important',
                    },
                },
            },
            MuiDialog: {
                styleOverrides: {
                    paperScrollPaper: {
                        backgroundColor: pal.mode === 'dark' ? undefined : colors.white0,
                    },
                },
            },
            MuiButtonBase: {
                defaultProps: {
                    disableRipple: true,
                },
            },
        },
        typography,
    });
}
export function createDarkTheme() {
    return createDDTheme({
        mode: 'dark',
        background: {
            default: '#202C33',
            paper: '#202C33',
        },
        text: {
            primary: '#FFF',
            secondary: '#70787D',
            disabled: '#B7BBBE',
        },
    });
}
export function createLightTheme() {
    return createDDTheme({
        mode: 'light',
        background: {
            default: '#F4F4F6',
            paper: '#F4F4F6',
        },
        text: {
            secondary: '#70787D',
            disabled: '#B7BBBE',
        },
    });
}
